FirebaseAuth/Sources/Swift/SystemService/AuthAPNSTokenManager.swift

```diff
@@ -68,18 +68,19 @@
       kAuthGlobalWorkQueue.asyncAfter(deadline: deadline) {
         // Only cancel if the pending callbacks remain the same, i.e., not triggered yet.
         if applicableCallbacks.count == self.pendingCallbacks.count {
-          self.callback(withToken: nil, error: nil)
+          self.callback(.failure(AuthErrorUtils.missingAppTokenError(underlyingError: nil)))
```

Comment:
With this `AuthErrorUtils.missingAppTokenError(underlyingError: nil)`, the user of the SDK has no means to know that the request timed out, apart from implementing a timeout logic on their side, right?

Would it be a solution to add an underlying timeout Error?

Reply:
Hi @snapxo, that's correct that the timeout error doesn't bubble up. This `getToken` API is called internally and in the case of an error (e.g. like from timeout), it actually attempts the recaptcha flow rather than bubble the error up to the user:

https://github.com/firebase/firebase-ios-sdk/blob/e76d7460aa3d19c5503a072f3117f69e2c443b11/FirebaseAuth/Sources/Swift/AuthProvider/PhoneAuthProvider.swift#L321-L326

Reply:
Makes sense if the fallback upon every `missingToken` error is the recaptcha 👌